scvi/model/_scanvi.py

```diff
@@ -198,37 +191,31 @@ def _data_loader_cls(self):
     @property
     def history(self):
         """Returns computed metrics during training."""
-        return {
-            "unsupervised_trainer_history": self.unsupervised_history_,
-            "semisupervised_trainer_history": self.semisupervised_history_,
-        }
+        return self._trainer.logger.history
 
     def train(
         self,
-        n_epochs_unsupervised: Optional[int] = None,
-        n_epochs_semisupervised: Optional[int] = None,
+        max_epochs: Optional[int] = None,
+        n_samples_per_label: Optional[float] = None,
         train_size: float = 0.9,
         validation_size: Optional[float] = None,
         batch_size: int = 512,
         use_gpu: Optional[bool] = None,
+        num_workers: int = 4,
```

Comment:
idk how I feel about num_workers being accessible here. I feel like it could be good if we (for the next PR)

1. Add `dl_num_workers` to our settings config with the pytorch default.
2. Catch the pytorch lightning user warning about it and create a more informative logger warning about it, as well as how to change it.
3. Add `pin_memory_if_gpu = True` to our settings.

Basically batch size is more of a training parameter like learning rate than the dataloader kwargs, and those things should be isolated IMO

Reply:
I added this to notion. For now, let's remove num_workers